event-handlers/src/main/java/no/sikt/nva/nvi/events/cristin/CristinMapper.java

```diff
@@ -24,37 +33,110 @@ public final class CristinMapper {
     public static final String GZ_EXTENSION = ".gz";
     public static final String PUBLICATION = "publication";
     public static final String RESOURCES = "resources";
+    public static final String PERSON = "person";
 
     private CristinMapper() {
 
     }
 
     public static DbCandidate toDbCandidate(CristinNviReport cristinNviReport) {
+        var now = Instant.now();
         return DbCandidate.builder()
                    .publicationId(constructPublicationId(cristinNviReport.publicationIdentifier()))
                    .publicationBucketUri(constructPublicationBucketUri(cristinNviReport.publicationIdentifier()))
-                   .publicationDate(constructPublicationDate(cristinNviReport))
+                   .publicationDate(constructPublicationDate(cristinNviReport.publicationDate()))
+                   .instanceType(cristinNviReport.instanceType())
+                   .creators(extractCreators(cristinNviReport))
+                   .level(extractLevel(cristinNviReport))
+                   .reportStatus(ReportStatus.REPORTED)
                    .applicable(true)
-                   .level(DbLevel.NON_CANDIDATE)
+                   .createdDate(now)
+                   .modifiedDate(now)
                    .build();
     }
 
+    public static List<DbCreator> extractCreators(CristinNviReport cristinNviReport) {
+        return cristinNviReport.scientificResources().get(0).getCreators().stream()
+                           .collect(groupByCristinIdentifierAndMapToInstitutionIdentifier())
+                           .entrySet().stream()
+                           .map(CristinMapper::toDbCreator)
+                           .toList();
+    }
+
+    private static DbCreator toDbCreator(Entry<URI, List<URI>> entry) {
+        return DbCreator.builder().creatorId(entry.getKey()).affiliations(entry.getValue()).build();
+    }
+
+    private static Collector<ScientificPerson, ?, Map<URI, List<URI>>> groupByCristinIdentifierAndMapToInstitutionIdentifier() {
```

Comment:
Rename to `groupByCristinIdentifierAndMapToAffiliationId`? To distinguish between top level institution and sub unit affiliations